MNIST example in the burn book needs a small modification

I upgraded to v0.14.0 and the MNIST example in the book has a small issue. `MnistBatcher<B>` has a function called `batch` and it has two variables, `images` and `targets`. I can see that the example is modified in one of the commits, and the `images` variable has no issue. However `targets` does not compile as is.

Below is the code that is in the documentation: 

```
let targets = items
      .iter()
      .map(|item| Tensor::<B, 1, Int>::from_data(
            TensorData::from([(item.label as i64).elem()]),
             &self.device
       ))
      .collect();
```

If compiled as is, I get the following error message: 

```
error[E0283]: type annotations needed
  --> src/conv/data.rs:41:21
   |
41 |                     TensorData::from([(item.label as i64).elem()]),
   |                     ^^^^^^^^^^ cannot infer type for array `[_; 1]`
   |
   = note: cannot satisfy `burn::tensor::TensorData: From<[_; 1]>`
   = help: the following types implement trait `From<T>`:
```

The way I fixed it is by modifying the `targets` variable to the following: 

```
let targets = items
    .iter()
    .map(|item| {
        Tensor::<B, 1, Int>::from_data(
            TensorData::from([item.label as i64]).convert::<B::IntElem>(),
            &self.device,
        )
    })
    .collect();
```

This might be a simple fix, but new users might find it challenging to do the modification on their. I wanted to bring it to your attention.

**To Reproduce**
Compile the MNIST example code from the documentation as is.

**Expected behavior**
The code should compile.

**Screenshots**

The error message: 

```
error[E0283]: type annotations needed
  --> src/conv/data.rs:41:21
   |
41 |                     TensorData::from([(item.label as i64).elem()]),
   |                     ^^^^^^^^^^ cannot infer type for array `[_; 1]`
   |
   = note: cannot satisfy `burn::tensor::TensorData: From<[_; 1]>`
   = help: the following types implement trait `From<T>`:
```

**Desktop (please complete the following information):**
 - Device: Macbook M1
 - Rust - Stable 1.80.1

--- a/burn-book/src/basic-workflow/data.md
+++ b/burn-book/src/basic-workflow/data.md
@@ -79,10 +79,12 @@ impl<B: Backend> Batcher<MnistItem, MnistBatch<B>> for MnistBatcher<B> {
 
         let targets = items
             .iter()
-            .map(|item| Tensor::<B, 1, Int>::from_data(
-                TensorData::from([(item.label as i64).elem()]),
-                &self.device
-            ))
+            .map(|item| {
+                Tensor::<B, 1, Int>::from_data(
+                    [(item.label as i64).elem::<B::IntElem>()],
+                    &self.device,
+                )
+            })
             .collect();
 
         let images = Tensor::cat(images, 0).to_device(&self.device);

--- a/burn-book/src/basic-workflow/inference.md
+++ b/burn-book/src/basic-workflow/inference.md
@@ -10,7 +10,7 @@ cost. Let's create a simple `infer` method in a new file `src/inference.rs` whic
 load our trained model.
 
 ```rust , ignore
-# use crate::{data::MnistBatcher, model::Model, training::TrainingConfig};
+# use crate::{data::MnistBatcher, training::TrainingConfig};
 # use burn::{
 #     data::{dataloader::batcher::Batcher, dataset::vision::MnistItem},
 #     prelude::*,

--- a/burn-book/src/basic-workflow/inference.md
+++ b/burn-book/src/basic-workflow/inference.md
@@ -10,15 +10,12 @@ cost. Let's create a simple `infer` method in a new file `src/inference.rs` whic
 load our trained model.
 
 ```rust , ignore
+# use crate::{data::MnistBatcher, model::Model, training::TrainingConfig};
 # use burn::{
-#     config::Config,
 #     data::{dataloader::batcher::Batcher, dataset::vision::MnistItem},
-#     module::Module,
+#     prelude::*,
 #     record::{CompactRecorder, Recorder},
-#     tensor::backend::Backend,
 # };
-# 
-# use crate::{data::MnistBatcher, training::TrainingConfig};
 # 
 pub fn infer<B: Backend>(artifact_dir: &str, device: B::Device, item: MnistItem) {
     let config = TrainingConfig::load(format!("{artifact_dir}/config.json"))

--- a/burn-book/src/basic-workflow/model.md
+++ b/burn-book/src/basic-workflow/model.md
@@ -221,8 +221,8 @@ impl ModelConfig {
 At a glance, you can view the model configuration by printing the model instance:
 
 ```rust , ignore
+use crate::model::ModelConfig;
 use burn::backend::Wgpu;
-use guide::model::ModelConfig;
 
 fn main() {
     type MyBackend = Wgpu<f32, i32>;

--- a/burn-book/src/basic-workflow/training.md
+++ b/burn-book/src/basic-workflow/training.md
@@ -39,7 +39,9 @@ impl<B: Backend> Model<B> {
         targets: Tensor<B, 1, Int>,
     ) -> ClassificationOutput<B> {
         let output = self.forward(images);
-        let loss = CrossEntropyLoss::new(None, &output.device()).forward(output.clone(), targets.clone());
+        let loss = CrossEntropyLossConfig::new()
+            .init(&output.device())
+            .forward(output.clone(), targets.clone());
 
         ClassificationOutput::new(loss, output, targets)
     }
@@ -60,26 +62,21 @@ Moving forward, we will proceed with the implementation of both the training and
 for our model.
 
 ```rust , ignore
+# use crate::{
+#     data::{MnistBatch, MnistBatcher},
+#     model::{Model, ModelConfig},
+# };
 # use burn::{
-#     config::Config,
 #     data::{dataloader::DataLoaderBuilder, dataset::vision::MnistDataset},
-#     module::Module,
-#     nn::loss::CrossEntropyLoss,
+#     nn::loss::CrossEntropyLossConfig,
 #     optim::AdamConfig,
+#     prelude::*,
 #     record::CompactRecorder,
-#     tensor::{
-#         backend::{AutodiffBackend, Backend},
-#         Int, Tensor,
-#     },
+#     tensor::backend::AutodiffBackend,
 #     train::{
 #         metric::{AccuracyMetric, LossMetric},
 #         ClassificationOutput, LearnerBuilder, TrainOutput, TrainStep, ValidStep,
 #     },
-# };
-# 
-# use crate::{
-#     data::{MnistBatch, MnistBatcher},
-#     model::{Model, ModelConfig},
 # };
 # 
 # impl<B: Backend> Model<B> {
@@ -89,8 +86,9 @@ for our model.
 #         targets: Tensor<B, 1, Int>,
 #     ) -> ClassificationOutput<B> {
 #         let output = self.forward(images);
-#         let loss =
-#             CrossEntropyLoss::new(None, &output.device()).forward(output.clone(), targets.clone());
+#         let loss = CrossEntropyLossConfig::new()
+#             .init(&output.device())
+#             .forward(output.clone(), targets.clone());
 # 
 #         ClassificationOutput::new(loss, output, targets)
 #     }
@@ -147,26 +145,21 @@ Book.
 Let us move on to establishing the practical training configuration.
 
 ```rust , ignore
+# use crate::{
+#     data::{MnistBatch, MnistBatcher},
+#     model::{Model, ModelConfig},
+# };
 # use burn::{
-#     config::Config,
 #     data::{dataloader::DataLoaderBuilder, dataset::vision::MnistDataset},
-#     module::Module,
-#     nn::loss::CrossEntropyLoss,
+#     nn::loss::CrossEntropyLossConfig,
 #     optim::AdamConfig,
+#     prelude::*,
 #     record::CompactRecorder,
-#     tensor::{
-#         backend::{AutodiffBackend, Backend},
-#         Int, Tensor,
-#     },
+#     tensor::backend::AutodiffBackend,
 #     train::{
 #         metric::{AccuracyMetric, LossMetric},
 #         ClassificationOutput, LearnerBuilder, TrainOutput, TrainStep, ValidStep,
 #     },
-# };
-# 
-# use crate::{
-#     data::{MnistBatch, MnistBatcher},
-#     model::{Model, ModelConfig},
 # };
 # 
 # impl<B: Backend> Model<B> {
@@ -176,8 +169,9 @@ Let us move on to establishing the practical training configuration.
 #         targets: Tensor<B, 1, Int>,
 #     ) -> ClassificationOutput<B> {
 #         let output = self.forward(images);
-#         let loss =
-#             CrossEntropyLoss::new(None, &output.device()).forward(output.clone(), targets.clone());
+#         let loss = CrossEntropyLossConfig::new()
+#             .init(&output.device())
+#             .forward(output.clone(), targets.clone());
 # 
 #         ClassificationOutput::new(loss, output, targets)
 #     }

